db/db_iter.cc

```diff
@@ -846,6 +846,7 @@ bool DBIter::FindValueForCurrentKey() {
   }
 
   Status s;
+  s.PermitUncheckedError();
```

Comment:
Not related to this PR, but I think a default initialized ```Status``` should ignore the status_checked flag. This pattern seems a little odd.

Reply:
Not arguing against it but the downside of doing it is that there might be some cases where we have something like this:

```
Status MyFunction() {
  Status s;
  if (<condition A>) {
    s = xxxx;
  }
  return s;
```
<condition A> is never covered in unit tests. Now at least it would raise our awareness that this return status is never checked.

Reply:
@siying @anand1976 I submitted a PR to address the "default initialized Status" case.  

For this specific function, I believe changing the kTypeMerge for "lastBlobIndex" case to set s (instead of status_) and not return will be identical behavior to what is there now and fix one case:

      } else if (last_not_merge_type == kTypeBlobIndex) {
        if (!allow_blob_) {
          ROCKS_LOG_ERROR(logger_, "Encounter unexpected blob index.");
          s = Status::NotSupported(
              "Encounter unexpected blob index. Please open DB with "
              "ROCKSDB_NAMESPACE::blob_db::BlobDB instead.");
        } else {
          s = Status::NotSupported("Blob DB does not support merge operator.");
        }

For the kBlobIndex, something similar can be done:

      if (!allow_blob_) {
        ROCKS_LOG_ERROR(logger_, "Encounter unexpected blob index.");
        s = Status::NotSupported(
            "Encounter unexpected blob index. Please open DB with "
            "ROCKSDB_NAMESPACE::blob_db::BlobDB instead.");
      } else {
          is_blob_ = true;
      }
     break;

Reply:
IIUC, @siying's point is that enforcing a check even for default initialized Status is useful for drawing attention to potential codepaths where a Status is not checked. Consider this -
```
bool foo() {
  Status s;
  switch (bar) {
     case A:
        s = xxxx;
        return false;
     case B:
        return true;
  }

  return true;
}
```
Clearly, the Status is being ignored in case A. If case A is not covered by any unit test, the status check assertion is never triggered.